data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py

```diff
@@ -0,0 +1,165 @@
+
+def getTemporaryInDataFileName(number):
+    return "psmInData" + str(number) + ".tab"
+
+def getTerminalCommand(args, data):
+    result = args.python_version
+    result += " " + testScriptPath
+    result += " -d " + data
+    if len(args.flags) > 0:
+        result += " -f \"" + args.flags + "\""
+    result += " -r " + str(args.runs)
+    if args.await_tests is True:
+        result += " -a "
+    print("RES: " + result)
+    return result
+    
+def addElement(dict, key, value):
+    if key not in dict:
+        dict[key] = []
+    dict[key].append(value)
+
+def getChartLinesShortened(tmpNumLines):
+    tmpNumLines /= 1000000
+    return int(tmpNumLines)
+
+def getBarChart(y_valuesWall, y_valuesCPU, x_numLines):
+    d1 = {}
+    d2 = {}
+    data = {}
+    for i in range (0,len(x_numLines)):
+        d1[x_numLines[i]] = y_valuesWall[i]
+        d2[x_numLines[i]] = y_valuesCPU[i]
+    data['Wall'] = d1
+    data['CPU'] = d2
+    df = pd.DataFrame(data)
+    df.plot(kind='bar', width=0.4, color=['darkgray','gray'])
+    colors = {'Wall':'darkgray', 'CPU':'gray'}         
+    labels = list(colors.keys())
+    handles = [plt.Rectangle((0,0),1,1, color=colors[label]) for label in labels]
+    plt.legend(handles, labels)
+    plt.tight_layout()
+    plt.setp(plt.xticks()[1], rotation=0)
+    return plt
+
+
+args = getArguments()
+testDataFilePath = args.data
+
+#################
+numLines = 5000000
+numDataFiles = 0
+numLinesMax = rawincount(testDataFilePath)
+for i in range(0, int(numLinesMax/numLines)):
+    numDataFiles += 1
+    makeReducedFile(testDataFilePath, testDataFolderPath + "/" + getTemporaryInDataFileName(i+1), numLines*numDataFiles)
+#################
+runTimes = []
+for i in range(1, numDataFiles+1):
+    terminalCommand = getTerminalCommand(args, testDataFolderPath + "/" + getTemporaryInDataFileName(i))
+    p = subprocess.Popen(terminalCommand, stderr=PIPE , stdout=PIPE , text=True, shell=True)
+    print("Running subprocess: " + str(p.pid))
+    p.wait()
+    result = p.stdout.read()
+    print(p.stderr.read())
+    result = result.splitlines()
+    for item in result:
+        if "Min" in item:
+            runTime = re.findall(r"[-+]?\d*\.\d+|\d+", item)
+            runTime = [float(i) for i in runTime]
+            runTimes.append(runTime)
+#################
+y_valuesWall = []
+y_valuesCPU = []
+x_numLines = []
+for i in range(1, numDataFiles+1):
+    numHandledLines = 0
+    if(i  == numDataFiles):
+        numHandledLines = numLinesMax
+    else:
+        numHandledLines = i*numLines
+    x_numLines.append(numHandledLines)
+    wallSec = runTimes[i-1][1]
+    cpuSec = runTimes[i-1][0]
+    y_valuesWall.append(wallSec)
+    y_valuesCPU.append(cpuSec)
+x_numLines = [getChartLinesShortened(item) for item in x_numLines]
+##############################
+chart = getBarChart(y_valuesWall, y_valuesCPU, x_numLines)
+chart.savefig(str(scriptDirectory) + "/" + args.filename + ".png")
```

Comment:
`scriptDirectory` is already a string. Why not use format here: `"{}/{}.png".format(scriptDirectory, args.filename)`
?